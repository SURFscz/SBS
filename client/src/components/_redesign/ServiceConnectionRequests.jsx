import React from "react";
import {ReactComponent as ChevronLeft} from "../../icons/chevron-left.svg";

import "./ServiceConnectionRequests.scss";
import "./UserColumn.scss";
import {isEmpty, stopEvent} from "../../utils/Utils";
import I18n from "../../locale/I18n";
import Button from "../button/Button";
import {setFlash} from "../../utils/Flash";
import ConfirmationDialog from "../confirmation-dialog/ConfirmationDialog";
import Entities from "./Entities";
import SpinnerField from "./SpinnerField";
import InputField from "../input-field/InputField";
import {approveServiceConnectionRequest, deleteServiceConnectionRequest, denyServiceConnectionRequest} from "../../api";
import moment from "moment";
import Logo from "./Logo";
import InstituteColumn from "./InstituteColumn";
import UserColumn from "./UserColumn";
import Select from "react-select";
import {Chip} from "@surfnet/sds";
import {chipTypeForStatus} from "../../utils/UserRole";

const allValue = "all";

class ServiceConnectionRequests extends React.Component {

    constructor(props, context) {
        super(props, context);
        this.state = {
            selectedServiceConnectionRequestId: null,
            confirmationDialogOpen: false,
            confirmationDialogQuestion: undefined,
            confirmationDialogAction: () => true,
            cancelDialogAction: () => this.setState({declineDialog: false, confirmationDialogOpen: false}),
            isWarning: false,
            declineDialog: false,
            loading: true,
            filterOptions: [],
            filterValue: {},
            rejectionReason: null
        }
    }

    componentDidMount = callback => {
        const {serviceConnectionRequests} = this.props;
        const filterOptions = [{
            label: I18n.t("collaborationRequest.statuses.all", {nbr: serviceConnectionRequests.length}),
            value: allValue
        }];
        const statusOptions = serviceConnectionRequests.reduce((acc, jr) => {
            const option = acc.find(opt => opt.status === jr.status);
            if (option) {
                ++option.nbr;
            } else {
                acc.push({status: jr.status, nbr: 1})
            }
            return acc;
        }, []).map(option => ({
            label: `${I18n.t("collaborationRequest.statuses." + option.status)} (${option.nbr})`,
            value: option.status
        })).sort((o1, o2) => o1.label.localeCompare(o2.label));

        this.setState({
            filterOptions: filterOptions.concat(statusOptions),
            filterValue: filterOptions[0],
            loading: false,
            declineDialog: false,
            selectedServiceConnectionRequestId: null
        }, callback);
    }

    refreshAndFlash = (promise, flashMsg, callback) => {
        this.closeConfirmationDialog();
        this.setState({loading: true});
        promise.then(res => {
            this.props.refresh(() => {
                this.componentDidMount(() => callback && callback(res));
                setFlash(flashMsg);
            });
        });
    }

    closeConfirmationDialog = () => this.setState({confirmationDialogOpen: false});

    getSelectedServiceConnectionRequest = () => {
        const {selectedServiceConnectionRequestId} = this.state;
        const {serviceConnectionRequests} = this.props;
        return serviceConnectionRequests.find(scr => scr.id === selectedServiceConnectionRequestId);
    }

    removeServiceConnectionRequest = () => {
        const {service} = this.props;
        const serviceConnectionRequest = this.getSelectedServiceConnectionRequest();
        const name = isEmpty(service) ? serviceConnectionRequest.service.name : service.name;
        this.confirm(() => {
            this.refreshAndFlash(deleteServiceConnectionRequest(serviceConnectionRequest.id),
                I18n.t("serviceConnectionRequest.flash.deleted", {
                    name: name
                }), () => this.componentDidMount())
        }, I18n.t("serviceConnectionRequest.deleteConfirmation"), false, false);

    };

    approveServiceConnectionRequest = () => {
        const {service} = this.props;
        const serviceConnectionRequest = this.getSelectedServiceConnectionRequest();
        const name = isEmpty(service) ? serviceConnectionRequest.service.name : service.name;
        this.confirm(() => {
            this.refreshAndFlash(approveServiceConnectionRequest(serviceConnectionRequest),
                I18n.t("serviceConnectionRequest.flash.accepted", {
                    name: name
                }), () => this.componentDidMount())
        }, I18n.t("serviceConnectionRequest.approveConfirmation"), false, false);
    };

    denyServiceConnectionRequest = () => {
        const {service} = this.props;
        const serviceConnectionRequest = this.getSelectedServiceConnectionRequest();
        const name = isEmpty(service) ? serviceConnectionRequest.service.name : service.name;
        this.confirm(() => {
            const {rejectionReason} = this.state;
            this.refreshAndFlash(denyServiceConnectionRequest(serviceConnectionRequest, rejectionReason),
                I18n.t("serviceConnectionRequest.flash.declined", {
                    name: name
                }), () => this.componentDidMount())
        }, I18n.t("serviceConnectionRequest.declineConfirmation"), true, true);
    };

    filter = (filterOptions, filterValue) => {
        return (
            <div className="service-connection-request-filter">
                <Select
                    className={"service-connection-request-filter-select"}
                    value={filterValue}
                    classNamePrefix={"filter-select"}
                    onChange={option => this.setState({filterValue: option})}
                    options={filterOptions}
                    isSearchable={false}
                    isClearable={false}
                />
            </div>
        );
    }

    getDeclineRejectionOptions = rejectionReason => {
        return (
            <div className="rejection-reason-container">
                <label htmlFor="rejection-reason">{I18n.t("joinRequest.rejectionReason")}</label>
                <InputField value={rejectionReason}
                            multiline={true}
                            onChange={e => this.setState({rejectionReason: e.target.value})}/>
                <span className="rejection-reason-disclaimer">{I18n.t("joinRequest.rejectionReasonNote")}</span>
            </div>
        );
    }

    confirm = (action, question, isWarning, declineDialog) => {
        this.setState({
            confirmationDialogOpen: true,
            confirmationDialogQuestion: question,
            confirmationDialogAction: action,
            declineDialog: declineDialog,
            isWarning: isWarning
        });
    };

    cancelSideScreen = e => {
        stopEvent(e);
        this.setState({selectedServiceConnectionRequestId: null});
    }

    openServiceConnectionRequest = serviceConnectionRequest => e => {
        stopEvent(e);
        this.setState({selectedServiceConnectionRequestId: serviceConnectionRequest.id});
    };

    renderServiceConnectionRequest = (service, serviceConnectionRequest) => {
        const {
            confirmationDialogOpen,
            cancelDialogAction,
            confirmationDialogAction,
            confirmationDialogQuestion,
            isWarning,
            rejectionReason,
            declineDialog
        } = this.state;
        const isOpen = serviceConnectionRequest.status === "open"
        const isDeclined = serviceConnectionRequest.status === "denied";

        return (
            <div className="service-connection-requests-details-container">
                <ConfirmationDialog isOpen={confirmationDialogOpen}
                                    cancel={cancelDialogAction}
                                    isWarning={isWarning}
                                    disabledConfirm={declineDialog && isEmpty(rejectionReason)}
                                    confirm={confirmationDialogAction}
                                    question={confirmationDialogQuestion}>
                    {declineDialog && this.getDeclineRejectionOptions(rejectionReason)}
                </ConfirmationDialog>
                <div>
                    <a className={"back-to-service-connection-requests"}
                       onClick={this.cancelSideScreen}
                       href={"/#cancel"}>
                        <ChevronLeft/>{I18n.t("models.serviceConnectionRequests.backToServiceConnectionRequests")}
                    </a>
                </div>
                <div className="service-connection-request-form">
                    <div className="service-connection-request-header">
                        <h2>{I18n.t(`models.serviceConnectionRequests.${isEmpty(service) ? "detailsWithService" : "details"}`,
                            {
                                date: moment(serviceConnectionRequest.created_at * 1000).format("LL"),
                                name: serviceConnectionRequest.requester.name || serviceConnectionRequest.requester.uid,
                                collaborationName: serviceConnectionRequest.collaboration.name,
                                serviceName: isEmpty(service) ? serviceConnectionRequest.service.name : ""
                            })}</h2>
                        {<Chip label={I18n.t(`collaborationRequest.statuses.${serviceConnectionRequest.status}`)}
                               type={chipTypeForStatus(serviceConnectionRequest)}/>}
                    </div>

                    <InputField name={I18n.t("serviceConnectionRequest.message")}
                                value={serviceConnectionRequest.message}
                                disabled={true}
                                multiline={true}
                                toolTip={I18n.t("serviceConnectionRequest.messageTooltip", {name: serviceConnectionRequest.requester.name})}/>

                    {isDeclined && <InputField name={I18n.t("joinRequest.rejectionReasonLabel")}
                                               value={serviceConnectionRequest.rejection_reason}
                                               disabled={true}
                                               multiline={true}/>}

                    <section className="actions">
                        {isOpen && <Button cancelButton={true} txt={I18n.t("serviceConnectionRequest.decline")}
                                           onClick={this.denyServiceConnectionRequest}/>}
                        {isOpen && <Button txt={I18n.t("serviceConnectionRequest.accept")}
                                           onClick={this.approveServiceConnectionRequest}/>}
                        {!isOpen && <Button warningButton={true}
                                            onClick={this.removeServiceConnectionRequest}/>}
                    </section>
                </div>
            </div>)

    }

    render() {
        const {loading, filterOptions, filterValue} = this.state;
        const {service, serviceConnectionRequests, user: currentUser} = this.props;
        if (loading) {
            return <SpinnerField/>;
        }
        const selectedServiceConnectionRequest = this.getSelectedServiceConnectionRequest();
        if (selectedServiceConnectionRequest) {
            return this.renderServiceConnectionRequest(service, selectedServiceConnectionRequest);
        }
        const columns = [
            {
                nonSortable: true,
                key: "logo",
                header: "",
                mapper: serviceConnectionRequest => <Logo src={serviceConnectionRequest.collaboration.logo}/>
            },
            {
                key: "collaboration__name",
                class: isEmpty(service) ? "" : "has-service",
                header: I18n.t("models.serviceConnectionRequests.name"),
                mapper: serviceConnectionRequest => serviceConnectionRequest.collaboration.name
            },
            isEmpty(service) ? {
                key: "service__name",
                header: I18n.t("models.serviceConnectionRequests.serviceName"),
                mapper: serviceConnectionRequest => serviceConnectionRequest.service.name
            } : null,
            {
                key: "requester__name",
                class: isEmpty(service) ? "" : "has-service",
                header: I18n.t("models.serviceConnectionRequests.requester"),
                mapper: serviceConnectionRequest => <UserColumn entity={{user: serviceConnectionRequest.requester}}
                                                                currentUser={currentUser}/>
            },
            {
                key: "user__schac_home_organisation",
                header: I18n.t("models.users.institute"),
                mapper: serviceConnectionRequest => <InstituteColumn entity={{user: serviceConnectionRequest.requester}}
                                                                     currentUser={currentUser}/>
            },
            {
                key: "status",
                header: I18n.t("collaborationRequest.status"),
                mapper: entity => <Chip type={chipTypeForStatus(entity)}
                                        label={I18n.t(`collaborationRequest.statuses.${entity.status}`)}/>
            }
        ].filter(tab => tab !== null);

        const filteredServiceConnectionRequests = filterValue.value === allValue ? serviceConnectionRequests :
            serviceConnectionRequests.filter(jr => jr.status === filterValue.value);
        return (
            <div>
                <Entities entities={filteredServiceConnectionRequests}
                          modelName="serviceConnectionRequests"
                          searchAttributes={["requester__name", "requester__email", "message", "status", "collaboration__name"]}
                          defaultSort="collaboration__name"
                          rowLinkMapper={() => this.openServiceConnectionRequest}
                          columns={columns}
                          filters={this.filter(filterOptions, filterValue)}
                          loading={loading}
                          {...this.props}/>
            </div>
        )
    }
}

export default ServiceConnectionRequests;
