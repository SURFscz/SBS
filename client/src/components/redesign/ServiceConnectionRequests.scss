@import "../../stylesheets/vars";
@import "../../stylesheets/forms";
@import "../../stylesheets/unit_header";

table.serviceConnectionRequests {

    thead {
        th {
            &.collaboration__name {
                width: 40%;
                padding-left: 15px;
            }

            &.requester__name {
                width: 30%;
            }
            &.user__schac_home_organisation {
                width: 30%;
            }

        }
    }

    tbody {
        td {
            &.collaboration__name {
                padding-left: 15px;
            }

        }
    }
}

.service-connection-requests-details-container {
    display: flex;
    max-width: $medium;
    width: 100%;
    margin: 40px auto 25px auto;

    @media (max-width: $compact-medium) {
        flex-direction: column;
    }

    a.back-to-service-connection-requests {
        display: flex;
        align-items: flex-start;
        white-space: nowrap;
        color: $blue;
        text-decoration: underline;
        margin-right: 60px;


        svg {
            margin-right: 5px;
            fill: $blue;
            height: 22px;
        }
    }

    div.service-connection-request-form {
        width: 60%;
        @media (max-width: $compact-medium) {
            width: 100%;
            padding: 15px 15px 0 15px;
        }

        h1 {
            margin: 1px 0 30px 0;
        }

        display: flex;
        flex-direction: column;

        div.checkbox {
            margin-top: 25px;
        }
    }

}
