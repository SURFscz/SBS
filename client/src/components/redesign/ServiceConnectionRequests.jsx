import React from "react";
import {ReactComponent as ChevronLeft} from "../../icons/chevron-left.svg";

import "./ServiceConnectionRequests.scss";
import "./UserColumn.scss";
import {stopEvent} from "../../utils/Utils";
import I18n from "i18n-js";
import Button from "../Button";
import {setFlash} from "../../utils/Flash";
import ConfirmationDialog from "../ConfirmationDialog";
import Entities from "./Entities";
import SpinnerField from "./SpinnerField";
import InputField from "../InputField";
import {approveServiceConnectionRequestByHash, denyServiceConnectionRequestByHash} from "../../api";
import moment from "moment";
import Logo from "./Logo";
import InstituteColumn from "./InstituteColumn";
import UserColumn from "./UserColumn";

class ServiceConnectionRequests extends React.Component {

    constructor(props, context) {
        super(props, context);
        this.state = {
            selectedServiceConnectionRequestId: null,
            confirmationDialogOpen: false,
            confirmationDialogQuestion: undefined,
            confirmationDialogAction: () => true,
            cancelDialogAction: () => this.setState({confirmationDialogOpen: false}),
            isWarning: false,
            loading: true,
        }
    }

    componentDidMount = callback => {
        this.setState({loading: false, selectedServiceConnectionRequestId: null}, callback);
    }

    refreshAndFlash = (promise, flashMsg, callback) => {
        this.closeConfirmationDialog();
        this.setState({loading: true});
        promise.then(res => {
            this.props.refresh(() => {
                this.componentDidMount(() => callback && callback(res));
                setFlash(flashMsg);
            });
        });
    }

    closeConfirmationDialog = () => this.setState({confirmationDialogOpen: false});

    getSelectedServiceConnectionRequest = () => {
        const {selectedServiceConnectionRequestId} = this.state;
        const {serviceConnectionRequests} = this.props;
        return serviceConnectionRequests.find(scr => scr.id === selectedServiceConnectionRequestId);
    }

    approveServiceConnectionRequest = () => {
        const {service} = this.props;
        const serviceConnectionRequest = this.getSelectedServiceConnectionRequest();
        this.confirm(() => {
            this.refreshAndFlash(approveServiceConnectionRequestByHash(serviceConnectionRequest.hash),
                I18n.t("serviceConnectionRequest.flash.accepted", {
                    name: service.name
                }), () => this.componentDidMount())
        }, I18n.t("serviceConnectionRequest.approveConfirmation"), false);

    };

    denyServiceConnectionRequest = () => {
        const {service} = this.props;
        const serviceConnectionRequest = this.getSelectedServiceConnectionRequest();
        this.confirm(() => {
            this.refreshAndFlash(denyServiceConnectionRequestByHash(serviceConnectionRequest.hash),
                I18n.t("serviceConnectionRequest.flash.denied", {
                    name: service.name
                }), () => this.componentDidMount())
        }, I18n.t("serviceConnectionRequest.declineConfirmation"), true);
    };

    confirm = (action, question, isWarning) => {
        this.setState({
            confirmationDialogOpen: true,
            confirmationDialogQuestion: question,
            confirmationDialogAction: action,
            isWarning: isWarning
        });
    };

    cancelSideScreen = e => {
        stopEvent(e);
        this.setState({selectedServiceConnectionRequestId: null});
    }

    openServiceConnectionRequest = serviceConnectionRequest => e => {
        stopEvent(e);
        this.setState({selectedServiceConnectionRequestId: serviceConnectionRequest.id});
    };

    renderServiceConnectionRequest = (service, serviceConnectionRequest) => {
        const {
            confirmationDialogOpen,
            cancelDialogAction,
            confirmationDialogAction,
            confirmationDialogQuestion,
            isWarning
        } = this.state;
        return (
            <div className="service-connection-requests-details-container">
                <ConfirmationDialog isOpen={confirmationDialogOpen}
                                    cancel={cancelDialogAction}
                                    isWarning={isWarning}
                                    confirm={confirmationDialogAction}
                                    question={confirmationDialogQuestion}/>
                <div>
                    <a className={"back-to-service-connection-requests"} onClick={this.cancelSideScreen}
                       href={"/cancel"}>
                        <ChevronLeft/>{I18n.t("models.serviceConnectionRequests.backToServiceConnectionRequests")}
                    </a>
                </div>
                <div className="service-connection-request-form">
                    <h2>{I18n.t("models.serviceConnectionRequests.details",
                        {
                            date: moment(serviceConnectionRequest.created_at * 1000).format("LL"),
                            name: serviceConnectionRequest.requester.name || serviceConnectionRequest.requester.uid,
                            collaborationName: serviceConnectionRequest.collaboration.name
                        })}</h2>

                    <InputField name={I18n.t("serviceConnectionRequest.message")}
                                value={serviceConnectionRequest.message}
                                disabled={true}
                                multiline={true}
                                toolTip={I18n.t("serviceConnectionRequest.messageTooltip", {name: serviceConnectionRequest.requester.name})}/>

                    <section className="actions">
                        <Button cancelButton={true} txt={I18n.t("serviceConnectionRequest.decline")}
                                onClick={this.denyServiceConnectionRequest}/>
                        <Button txt={I18n.t("serviceConnectionRequest.accept")}
                                onClick={this.approveServiceConnectionRequest}/>
                    </section>
                </div>
            </div>)

    }

    render() {
        const {loading} = this.state;
        const {service, serviceConnectionRequests, user: currentUser} = this.props;
        if (loading) {
            return <SpinnerField/>;
        }
        const selectedServiceConnectionRequest = this.getSelectedServiceConnectionRequest();
        if (selectedServiceConnectionRequest) {
            return this.renderServiceConnectionRequest(service, selectedServiceConnectionRequest);
        }
        const columns = [
            {
                nonSortable: true,
                key: "logo",
                header: "",
                mapper: serviceConnectionRequest => <Logo src={serviceConnectionRequest.collaboration.logo}/>
            },
            {
                key: "collaboration__name",
                header: I18n.t("models.serviceConnectionRequests.name"),
                mapper: serviceConnectionRequest => serviceConnectionRequest.collaboration.name
            },
            {
                key: "requester__name",
                header: I18n.t("models.serviceConnectionRequests.requester"),
                mapper: serviceConnectionRequest => <UserColumn entity={{user:serviceConnectionRequest.requester}} currentUser={currentUser}/>
            },
            {
                key: "user__schac_home_organisation",
                header: I18n.t("models.users.institute"),
                mapper: serviceConnectionRequest => <InstituteColumn entity={{user:serviceConnectionRequest.requester}} currentUser={currentUser}/>
            },

        ]
        return (
            <div>
                <Entities entities={serviceConnectionRequests}
                          modelName="serviceConnectionRequests"
                          searchAttributes={["requester__name", "requester__email", "message", "collaboration__name"]}
                          defaultSort="collaboration__name"
                          rowLinkMapper={() => this.openServiceConnectionRequest}
                          columns={columns}
                          loading={loading}
                          hideTitle={true}
                          {...this.props}/>
            </div>
        )
    }
}

export default ServiceConnectionRequests;