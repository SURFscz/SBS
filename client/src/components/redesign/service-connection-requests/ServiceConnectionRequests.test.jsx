import React from "react";
import { render } from "@testing-library/react";
import ServiceConnectionRequests from "./ServiceConnectionRequests";

describe("ServiceConnectionRequests", () => {
    it("matches snapshot with default props", () => {
        const { asFragment } = render(
            <ServiceConnectionRequests
                serviceConnectionRequests={[]}
                onAccept={() => {}}
                onReject={() => {}}
                user={{ admin: true }}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
