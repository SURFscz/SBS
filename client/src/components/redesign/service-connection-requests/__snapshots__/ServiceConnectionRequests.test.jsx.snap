// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceConnectionRequests matches snapshot with default props 1`] = `
<DocumentFragment>
  <div>
    <div
      class="mod-entities "
    >
      <section
        class="entities-search"
      >
        <h2>
          Connection requests (0)
        </h2>
        <div
          class="search-filters filters-with-title"
        >
          <div
            class="service-connection-request-filter"
          >
            <div
              class="service-connection-request-filter-select css-b62m3t-container"
            >
              <span
                class="css-1f43avz-a11yText-A11yText"
                id="react-select-2-live-region"
              />
              <span
                aria-atomic="false"
                aria-live="polite"
                aria-relevant="additions text"
                class="css-1f43avz-a11yText-A11yText"
                role="log"
              />
              <div
                class="filter-select__control css-13cymwt-control"
              >
                <div
                  class="filter-select__value-container filter-select__value-container--has-value css-hlgwow"
                >
                  <div
                    class="filter-select__single-value css-1dimb5e-singleValue"
                  >
                    All (0)
                  </div>
                  <input
                    aria-activedescendant=""
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-readonly="true"
                    class="css-1s80ejz-dummyInput-DummyInput"
                    id="react-select-2-input"
                    inputmode="none"
                    role="combobox"
                    tabindex="0"
                    value=""
                  />
                </div>
                <div
                  class="filter-select__indicators css-1wy0on6"
                >
                  <span
                    class="filter-select__indicator-separator css-1u9des2-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class="filter-select__indicator filter-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="search standalone"
        >
          <div
            class="sds--text-field sds--text-field--has-icon"
          >
            <div
              class="sds--text-field--shape"
            >
              <div
                class="sds--text-field--input-and-icon"
              >
                <input
                  class="sds--text-field--input"
                  placeholder="Search for connection requests..."
                  type="search"
                  value=""
                />
                <span
                  class="sds--text-field--icon"
                >
                  <svg>
                    search.svg
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="entities-list"
      />
      <div />
    </div>
  </div>
</DocumentFragment>
`;
